e2e/testenv/infra/go.mod

```diff
@@ -7,7 +7,7 @@ toolchain go1.22.0
 replace k8s.io/client-go => k8s.io/client-go v0.29.0
 
 require (
-	github.com/rancher/fleet v0.9.0-rc.4.0.20240208111910-ad2d2d829a12
```

Comment:
let's bump to `[v0.10.0-rc.5](https://github.com/rancher/fleet/releases/tag/v0.10.0-rc.5)` instead